packages/pigeon/lib/pigeon_lib.dart

```diff
@@ -632,24 +642,27 @@ class _RootBuilder extends dart_ast_visitor.RecursiveAstVisitor<Object?> {
     return null;
   }
 
+  Field formalParameterToField(dart_ast.FormalParameter parameter) {
+    final dart_ast.TypeName typeName = parameter.childEntities
+        // ignore: always_specify_types
```

Comment:
Why is this ignoring rather than following the repo style?

Reply:
Because the type isn't visible to users of the analyzer.  I've already filed a bug with them about all these datatypes we can reference but not name.

Reply:
> Because the type isn't visible to users of the analyzer.

Can you elaborate on what this means? I just checked this out locally; the IDE told me its type was `SyntacticEntity`, and I had no issue adding that type once I added the relevant import (`'package:analyzer/dart/ast/syntactic_entity.dart'`)

Reply:
What version of dart are you on? On stable 2.13.3 you can only get `SyntacticEntity` from `import 'package:_fe_analyzer_shared/src/base/syntactic_entity.dart';`  They may have fixed it in the version you are using.

Reply:
FWIW here is the other case where you can access a type but not reference a name, it turned out it was for `Source` not `SyntacticEntity`: https://github.com/dart-lang/sdk/issues/46420  There is also a 2.0 version of the analyzer, you may have been using that.

Reply:
It's showing analyzer 1.7.2

Reply:
If for some reason we can't use whatever version has that fix, you could at least make it `Object` with a TODO instead of turning off the lint.

Reply:
It looks like it does exist, DartCode didn't want to find it though, I added a bug: https://github.com/Dart-Code/Dart-Code/issues/3513

I'm not sure why they put it in its own module.  It should be in `dart_ast.dart`.